tools/IoTEdgeDevOps/DevOpsLib/BugQuery.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft. All rights reserved.
+using System;
+
+namespace DevOpsLib
+{
+    public class BugQuery
+    {
+        public BugQuery(string title, string area, BugPriority priority, bool inProgress)
+        {
+            ValidationUtil.ThrowIfNullOrWhiteSpace(title, nameof(title));
+            ValidationUtil.ThrowIfNullOrWhiteSpace(area, nameof(area));
+            ValidationUtil.ThrowIfNull(inProgress, nameof(inProgress));
+
+            this.Title = title;
+            this.Area = area;
+            this.Priority = priority;
+            this.InProgress = inProgress;
+        }
+        public string Title { get; }
+        public string Area { get; }
+        public BugPriority Priority { get; }
+        public bool InProgress { get; }
+
+        public string GetWiqlFromConfiguration()
+        {
+            string query = "";
+
+            if (this.Priority != BugPriority.Other && !this.InProgress)
```

Comment:
Define a data collection to store the mapping to get BugWiqlQueries string and replace this if-else chain block.
1. Dictionary<BugPriority, Dictionary<bool, string>> or
2. Dictionary<Tuple<BugPriority,bool>, string>

Reply:
I feel like this would work if we weren't grouping multiple different <Priority, InProgress> states to the same wiql template. Doing it the way you described means we have to enumerate every combination. With the if else chain, multiple different configs can evaluate to the same template.